### PR TITLE
Adds Slash Commands Permissions

### PR DESCRIPTION
**Pull request by Shadorc**
**Description:** Adds Slash Commands Permissions.

**Justification:** 
https://github.com/Discord4J/discord-json/pull/52
https://github.com/discord/discord-api-docs/pull/2737

### CODE REVIEW

**Comment by quanticc** [comment]
Will need an update to discord-json version at properties file

**Comment by Shadorc** [comment]
This will be ready to review once https://github.com/Discord4J/discord-json/pull/71/files is merged.

**Comment by Shadorc** [comment]
Build is passing locally but it seems that CI encounters "javadoc: error - Error fetching URL: https://projectreactor.io/docs/core/release/api/"

File: rest/src/main/java/discord4j/rest/service/ApplicationService.java [hunk 1]
```diff
@@ -119,4 +121,36 @@ public ApplicationService(Router router) {
...
+        return Routes.APPLICATION_COMMAND_PERMISSIONS_GET.newRequest(applicationId, guildId, commandId)
+                .exchange(getRouter())
+                .bodyToMono(GuildApplicationCommandPermissionsData.class);
+    }
+
+    public Mono<Void> modifyApplicationCommandPermissions(long applicationId, long guildId,
+                                                          long commandId,
+                                                          List<ApplicationCommandPermissionsData> permissions) {
+        return Routes.APPLICATION_COMMAND_PERMISSIONS_MODIFY.newRequest(applicationId, guildId, commandId)
+                .body(permissions)
```
Comment (darichey) [inline rest/src/main/java/discord4j/rest/service/ApplicationService.java:144]: I think we need a wrapper around this field. The docs say the body is an object with a `permissions` field, not just the array.

**Review by darichey — CHANGES_REQUESTED** [review]

File: rest/src/main/java/discord4j/rest/service/ApplicationService.java (hunk 1 above)
Comment (Shadorc) [reply rest/src/main/java/discord4j/rest/service/ApplicationService.java:144]: You're right, I have created a PR to implement the wrapper: https://github.com/Discord4J/discord-json/pull/75/files

**Review by darichey — APPROVED** [review]

**Pull request merged**